src/iris/sender/auditlog.py

```diff
@@ -15,11 +16,27 @@ def message_change(message_id, change_type, old, new, description):
         logger.warn('Not logging %s for message as it does not have an id', change_type)
         return
 
-    session = db.Session()
-    session.execute('''
-      INSERT INTO `message_changelog` (`message_id`, `change_type`, `old`, `new`, `description`, `date`)
-      VALUES (:message_id, :change_type, :old, :new, :description, NOW())
-    ''', dict(message_id=message_id, change_type=change_type, old=old, new=new, description=description))
-    session.commit()
-    session.close()
-    logger.info('Logged change information for message (ID %s)', message_id)
+    # retry to guard against deadlocks
+    retries = 0
+    max_retries = 5
+    while True:
+        with db.guarded_session() as session:
```

Comment:
We shouldn't use both this `db.guarded_session()` context manager as well as `db.Session()` on line 26 to instantiate a session, as only the session created on line 26 is getting used.

Reply:
fixed